presto-main/src/main/java/io/prestosql/sql/planner/DomainTranslator.java

```diff
@@ -928,7 +931,55 @@ protected ExtractionResult visitLikePredicate(LikePredicate node, Boolean comple
             }
 
             Slice constantPrefix = LikeFunctions.unescapeLiteralLikePattern(pattern.slice(0, patternConstantPrefixBytes), escape);
+            return createRangeDomain(type, constantPrefix).map(domain -> new ExtractionResult(TupleDomain.withColumnDomains(ImmutableMap.of(symbol, domain)), node));
+        }
+
+        @Override
+        protected ExtractionResult visitFunctionCall(FunctionCall node, Boolean complement)
+        {
+            String name = ResolvedFunction.extractFunctionName(node.getName());
+            if (name.equals("starts_with")) {
+                Optional<ExtractionResult> result = tryVisitStartsWithFunction(node, complement);
+                if (result.isPresent()) {
+                    return result.get();
+                }
+            }
+            return visitExpression(node, complement);
+        }
+
+        private Optional<ExtractionResult> tryVisitStartsWithFunction(FunctionCall node, Boolean complement)
+        {
+            List<Expression> args = node.getArguments();
+            if (args.size() != 2) {
+                return Optional.empty();
+            }
+            if (!(args.get(0) instanceof SymbolReference)) {
```

Comment:
please extarct `args.get(0)` and `args.get(1)` as separate variables

Reply:
fixed.